core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -565,6 +565,11 @@ private void planDistribution(PlanRoot plan)
         }
 
         queryScheduler.set(scheduler);
+        stateMachine.addQueryInfoStateChangeListener(queryInfo -> {
+            if (queryInfo.isFinalQueryInfo()) {
```

Comment:
Could query info be set without calling ` buildQueryInfo`? If so, how?

Reply:
Screenshot from Azul Mission Control before the change, after query sequential execution (at most one query running). It can clearly be seen that historical queries are holding on to query scheduler.

<img width="1186" alt="image" src="https://github.com/trinodb/trino/assets/144328/3e651920-5e4a-4471-8d9b-d2e9c49850ba">